compiler/rustc_target/src/spec/base/lynxos_178.rs

```diff
@@ -0,0 +1,152 @@
+use std::collections::BTreeMap;
+use std::env;
+use std::path::{Path, PathBuf};
+
+use crate::spec::{
+    Cc, CheckEnvironment, LinkerFlavor, Lld, PanicStrategy, RelocModel, RelroLevel, SplitDebuginfo,
+    StackProbeType, TargetOptions, cvs,
+};
+
+/// Strip mutability from &mut str.
+fn immutable(value: &mut str) -> &str {
+    let immutable = value;
+    immutable
+}
+
+macro_rules! cow_format {
+    ($arg:expr) => {
+        // Cow::from works with &str but not with &mut str.
+        Cow::from(immutable(Box::leak(format!($arg).into_boxed_str())))
+    };
+}
+
+struct EnvVar<'a> {
+    name: &'a str,
+    description: &'a str,
+}
+
+static ENV_PREFIX_VAR: EnvVar<'_> =
+    EnvVar { name: "ENV_PREFIX", description: "your LynxOS-178 Environment directory" };
+static GCC_DIR_VAR: EnvVar<'_> =
+    EnvVar { name: "GCC_DIR", description: "your LynxOS-178 GCC compiler directory" };
+
+struct LynxosPath<'a> {
+    env_var: &'a EnvVar<'a>,
+    rel_path: &'a str,
+}
+
+impl LynxosPath<'_> {
+    fn path(&self) -> Result<PathBuf, String> {
+        let var_value = env::var(self.env_var.name).map_err(|_| {
+            format!("Set {} to point at {}.", self.env_var.name, self.env_var.description)
+        })?;
+        Ok(Path::new(var_value.as_str()).join(self.rel_path))
+    }
+}
+
+// Paths relative to ENV_PREFIX
+static LINK_DIRS: [LynxosPath<'static>; 3] = [
+    LynxosPath { env_var: &GCC_DIR_VAR, rel_path: "" },
+    LynxosPath { env_var: &ENV_PREFIX_VAR, rel_path: "lib/" },
+    LynxosPath { env_var: &ENV_PREFIX_VAR, rel_path: "usr/lib/" },
+];
+static LINK_LIBS: [LynxosPath<'static>; 5] = [
+    LynxosPath { env_var: &ENV_PREFIX_VAR, rel_path: "lib/crt1.o" },
+    LynxosPath { env_var: &ENV_PREFIX_VAR, rel_path: "lib/crti.o" },
+    LynxosPath { env_var: &GCC_DIR_VAR, rel_path: "crtbegin.o" },
+    LynxosPath { env_var: &GCC_DIR_VAR, rel_path: "crtend.o" },
+    LynxosPath { env_var: &ENV_PREFIX_VAR, rel_path: "lib/crtn.o" },
+];
+
+pub(crate) fn check_environment() -> Result<(), String> {
+    for rel_dir in &LINK_DIRS {
+        let dir = rel_dir.path()?;
+        if !dir.is_dir() {
+            return Err(format!(
+                "{} is not a directory. Does {} point at {}?",
+                dir.display(),
+                rel_dir.env_var.name,
+                rel_dir.env_var.description
+            ));
+        }
+    }
+
+    for rel_lib in &LINK_LIBS {
+        let lib = rel_lib.path()?;
+        if !lib.is_file() {
+            return Err(format!(
+                "{} is not a file. Does {} point at {}?",
+                lib.display(),
+                rel_lib.env_var.name,
+                rel_lib.env_var.description
+            ));
+        }
+    }
+
+    Ok(())
+}
+
+pub(crate) fn opts() -> TargetOptions {
+    let linker_flavor = LinkerFlavor::Gnu(Cc::No, Lld::No);
+    let other_linker_flavor = LinkerFlavor::Gnu(Cc::No, Lld::Yes);
+    let env_prefix = env::var("ENV_PREFIX").unwrap_or("ENV_PREFIX".into());
```

Comment:
Requiring `ENV_PREFIX` to be set won't work when just loading targets without the intention of compiling got LynxOS. You either need a dummy fallback or preferably not have it at all.
How is your toolchain expected to be used that introduces these requirements? I assume you have the same env var for C code already today.